app/models/experimentation/notifications_experiment.rb

```diff
@@ -42,34 +42,57 @@ def self.eligible_patients
                                              "end_time > ? OR treatment_group_memberships.created_at > ?",
                                              Time.current, MONITORING_BUFFER.ago
                                            ).select(:patient_id))
-        .then { |patients| exclude_bangladesh_blocks(patients) }
-        .then { |patients| include_only_allowed_india_states(patients) }
+        .then { |patients| filter_patients(patients, filters) }
     end
 
-    def self.exclude_bangladesh_blocks(patients)
-      return patients unless CountryConfig.current_country?("Bangladesh")
-
+    def self.filter_patients(patients, filters)
       patients
         .merge(Facility.with_block_region_id)
         .select("patients.*")
-        .where.not(block_region: {id: excluded_block_ids.presence})
+        .then { |patients| filter_states(patients, filters["states"]) }
+        .then { |patients| filter_blocks(patients, filters["blocks"]) }
+        .then { |patients| filter_facilities(patients, filters["facilities"]) }
     end
 
-    def self.include_only_allowed_india_states(patients)
-      return patients unless CountryConfig.current_country?("India") && SimpleServer.env.production?
-      return patients unless ENV["EXPERIMENT_ALLOWED_STATES"].present?
+    def self.filter_states(patients, states)
+      return patients unless states
 
-      states = ENV.fetch("EXPERIMENT_ALLOWED_STATES", "").split(",").map(&:strip)
-      patients.where(facilities: {state: states})
+      if states["include"]
+        patients.where(facilities: {state: states["include"]})
+      elsif states["exclude"]
+        patients.where.not(facilities: {state: states["exclude"]})
+      else
+        patients
+      end
     end
 
-    def self.excluded_block_ids
-      ENV.fetch("EXPERIMENT_EXCLUDED_BLOCKS", "").split(",").map(&:strip)
+    def self.filter_blocks(patients, blocks)
+      return patients unless blocks
+
+      if blocks["include"]
+        patients.where(block_region: {id: blocks["include"]})
+      elsif blocks["exclude"]
+        patients.where.not(block_region: {id: blocks["exclude"]})
+      else
+        patients
+      end
+    end
+
+    def self.filter_facilities(patients, facilities)
+      return patients unless facilities
+
+      if facilities["include"]
+        patients.where('"patients"."assigned_facility_id" IN (?)', facilities["include"])
+      elsif facilities["exclude"]
```

Comment:
qq: Can you not have facilities that you want to keep but some facilities that you want to exclude?